pkg/auth/spire/delegate.go

```diff
@@ -194,6 +200,21 @@ func (s *SpireDelegateClient) handleX509SVIDUpdate(svids []*delegatedidentityv1.
 	s.svidStoreMutex.Lock()
 	s.svidStore = newSvidStore
 	s.svidStoreMutex.Unlock()
+
+	for _, key := range updatedKeys {
+		// we send an update event to re-trigger a handshake if needed
+		id, err := s.spiffeIDToNumericIdentity(key)
+		if err != nil {
+			s.log.WithError(err).Errorf("failed to convert spiffe ID %s to numeric identity", key)
+			continue
+		}
+		select {
+		case s.rotatedIdentitiesChan <- certs.CertificateRotationEvent{Identity: id}:
+			s.log.Debugf("X509-SVID for %s has changed, signaling this", key)
+		default:
+			s.log.Warnf("skipping sending rotated identity %d as channel is full", id)
```

Comment:
might be worth to use the loglimiter to prevent flooding the log.

-> https://github.com/cilium/cilium/blob/master/pkg/auth/monitor/monitor.go#L72-L75